fw/sock_clnt.c

```diff
@@ -325,14 +313,16 @@ tfw_h2_sk_prepare_xmit(struct sock *sk, struct sk_buff *skb, unsigned int mss_no
 		if (unlikely(r)) {
 			T_WARN("%s: failed to make data frames (%d)",
 			       __func__, r);
-			r = (r == -ENOMEM && headers_was_done ? 0 : r);
-			goto update_limit;
+			if (r == -ENOMEM && headers_was_done) {
+				r = 0;
+				goto update_limit;
+			}
+			goto ret;
 		}
 
 		truesize += tmp_truesize;
 		tmp_truesize = 0;
 
-		skb_set_tfw_flags(skb, SS_F_HTTT2_FRAME_DATA_DONE);
```

Comment:
Can we drop SS_F_HTTT2_FRAME_HEADERS_DONE also?

Reply:
Done